csharp-intermediate-topics/WhenToUseReaderWriterLockSlimOverLockInCSharp/WhenToUseReaderWriterLockSlimOverLockInCSharp/BaseReaderWriter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Diagnostics;
+
+namespace WhenToUseReaderWriterLockSlimOverLockInCSharp
+{
+    public abstract class BaseReaderWriter
+    {
+        private readonly Stopwatch _stopwatch = new();
+        protected readonly List<int> _listOfNumbers = new();
+
+        public List<int> ListOfNumbers
+        {
+            get { return _listOfNumbers; }
+        }
+
+        protected int GetNextRandom()
+        {
+            var maxNumber = _listOfNumbers.Count > 0 ? _listOfNumbers.Count - 1 : 0;
+            return Random.Shared.Next(0, maxNumber);
+        }
+
+        public long Execute(ThreadExecutionConfiguration config)
```

Comment:
This method should still be pulled into its own thing. It is not related to the reader/writer, it executes code that uses the reader/writer. This is what I had mentioned previously and then created the fiddle to show an example of how to do it. This simplifies the code and keeps things separated/compartmentalized. Think of the "single responsibility" principle.

Reply:
put it in methods